src_c/simd_blitters_avx2.c

```diff
@@ -928,3 +928,174 @@ blit_blend_rgb_min_avx2(SDL_BlitInfo *info)
+                        27, 0x80, 27, 0x80, 27, 0x80, 15, 0x80, 15, 0x80, 15,
+                        0x80, 15, 0x80, 11, 0x80, 11, 0x80, 11, 0x80, 11);
+
+    mm256_ones = _mm256_set_epi8(
+        0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01,
+        0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01,
+        0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01);
+
+    while (height--) {
+        if (pre_8_width > 0) {
+            /* one pixel at a time - same as current sse2 version */
+            LOOP_UNROLLED4(
+                {
+                    Uint32 alpha = *srcp & amask;
+                    if (alpha == 0) {
+                        /* do nothing */
+                    }
+                    else if (alpha == amask) {
+                        *dstp = *srcp;
+                    }
+                    else {
+                        mm_src = _mm_cvtsi32_si128(*srcp);
+                        /*mm_src = 0x000000000000000000000000AARRGGBB*/
+                        mm_src = _mm_unpacklo_epi8(mm_src, mm_zero);
+                        /*mm_src = 0x000000000000000000AA00RR00GG00BB*/
+                        mm_dst = _mm_cvtsi32_si128(*dstp);
+                        /*mm_dst = 0x000000000000000000000000AARRGGBB*/
+                        mm_dst = _mm_unpacklo_epi8(mm_dst, mm_zero);
+                        /*mm_dst = 0x000000000000000000AA00RR00GG00BB*/
+
+                        mm_alpha = _mm_cvtsi32_si128(alpha);
+                        /* alpha -> mm_alpha (000000000000A000) */
+                        mm_alpha = _mm_srli_si128(mm_alpha, 3);
+                        /* mm_alpha >> ashift -> mm_alpha(000000000000000A) */
+                        mm_alpha = _mm_unpacklo_epi16(mm_alpha, mm_alpha);
+                        /* 0000000000000A0A -> mm_alpha */
+                        mm_alpha = _mm_unpacklo_epi32(mm_alpha, mm_alpha);
+                        /* 000000000A0A0A0A -> mm_alpha2 */
+
+                        /* pre-multiplied alpha blend */
+                        mm_sub_dst = _mm_add_epi16(mm_dst, mm_ones);
+                        mm_sub_dst = _mm_mullo_epi16(mm_sub_dst, mm_alpha);
+                        mm_sub_dst = _mm_srli_epi16(mm_sub_dst, 8);
+                        mm_dst = _mm_add_epi16(mm_src, mm_dst);
+                        mm_dst = _mm_sub_epi16(mm_dst, mm_sub_dst);
+                        mm_dst = _mm_packus_epi16(mm_dst, mm_zero);
+
+                        *dstp = _mm_cvtsi128_si32(mm_dst);
+                    }
+
+                    srcp += srcpxskip;
+                    dstp += dstpxskip;
+                },
+                n, pre_8_width);
+        }
+        srcp256 = (__m256i *)srcp;
+        dstp256 = (__m256i *)dstp;
+        if (post_8_width > 0) {
+            /*8 pixels at a time, need to use shuffle to get everything
+                lined up - see mul for an example*/
```

Comment:
It's crazy that the current SSE version is 1 pixel at a time and the AVX version is 8 pixels at a time, since the AVX registers are only twice as wide.